desc/external/paraview.py

```diff
@@ -0,0 +1,260 @@
+    res : tuple, list
+        The resolution used to create the mesh. Must be in the form
+        (N_toroidal, N_poloidal). Default is (100, 100).
+    keys : list, optional
+        The names of the quantities to be computed on the grid points. Defaults to
+        empty list.
+    rho : float
+        The flux surface to be exported if `obj` is an Equilibrium. Defaults to 1.
+    filename : str
+        Name for the saved file. The file extension will be `.vtu`. Default name will be
+        `surface.vtu`
+
+    Returns
+    -------
+    mesh : pyvista.UnstructuredGrid
+        Created unstructured grid object. With this object one can compute more
+        complicated quantities on `LinearGrid(rho=rho, theta=Np, zeta=Nt, NFP=1)`
+        and add it to the mesh by `mesh['complicated name'] = value`. Once the mesh
+        is changed, the user has to save it again `mesh.save(filename + ".vtu")`.
+    """
+    if not isinstance(obj, (Equilibrium, FourierRZToroidalSurface)):
+        raise ValueError(
+            "This function only support Equilibrium or FourierRZToroidalSurface "
+            f"objects but {type(obj)} is given."
+        )
+
+    Nt, Np = res
+    grid = LinearGrid(rho=rho, theta=Np, zeta=Nt, NFP=1)
+
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore", message="Unequal number of field")
+        data = obj.compute(["x"] + keys, grid=grid, basis="xyz")
+    nodes = np.asarray(data.pop("x"))
+
+    # Build connectivity: each quad is made of 4 points
+    cells = []
+    celltypes = []
+
+    for i in range(Nt):
+        for j in range(Np):
+            # Current point index
+            p0 = i * Np + j
+            # Neighbor indices (with wrap-around)
+            p1 = i * Np + (j + 1) % Np
+            p2 = ((i + 1) % Nt) * Np + (j + 1) % Np
+            p3 = ((i + 1) % Nt) * Np + j
+
+            # Append one quad: format = [4, pt0, pt1, pt2, pt3]
+            cells.extend([4, p0, p1, p2, p3])
+            celltypes.append(CellType.QUAD)
+
+    # Convert to numpy arrays
+    cells = np.array(cells)
+    celltypes = np.array(celltypes, dtype=np.uint8)
+
+    # Create the unstructured mesh
+    mesh = pv.UnstructuredGrid(cells, celltypes, nodes)
+
+    # Additional data
+    for key in keys:
+        value = data[key]
+        if len(value) == Nt * Np or value.shape == (Nt * Np, 3):
+            # this will handle both 1D and 3D
+            mesh[key] = value
+        else:
+            raise ValueError(f"Data {key} has an unimplemented shape ({value.shape})")
+
+    mesh.save(filename + ".vtu")
+    print(f"File is saved as {filename}.vtu")
+    return mesh
+
+
+def export_coils_to_paraview(coils, res=100, keys=[], filename="coil"):
+    """Export coils as VTP files.
+
+    Parameters
+    ----------
+    coils : Coil, CoilSet, MixedCoilSet
+        Object to be exported. Can be a single coil or multiple coils given
+        as CoilSet or MixedCoilSet.
+    res : int
```

Comment:
be clear that this is e.g. the resolution used to discretize each individual coil (and that the same res is used for every coil)

Also this res is the N for LinearGrid so this is not clear that the number of points used will be 2*res+1

either change docstring or just allow grid to be passed in with a default value set by N=100 if grid is None